yarl/_quoting_c.pyx

```diff
@@ -91,7 +93,15 @@ cdef struct Writer:
 
 
 cdef inline void _init_writer(Writer* writer):
-    writer.buf = &BUFFER[0]
+    cdef char *buf
+    if IS_GIL_DISABLED:
+        buf = <char *> PyMem_Malloc(BUF_SIZE)
+        if buf == NULL:
+            PyErr_NoMemory()
+            return
+        writer.buf = buf
```

Comment:
Why is this branch marked as not covered? https://app.codecov.io/gh/aio-libs/yarl/commit/dcb3ae1102d49a77b4bd824839c9a72c80cc229c?flags%5B0%5D=pytest&dropdown=coverage#fec01998dabbadc38df09608935ec4b7-R98

Reply:
@lysnikolaou this might be because the test job installs older cython: https://github.com/aio-libs/yarl/actions/runs/13526871108/job/37800294520?pr=1456#step:8:3. Could you try updating it there to see if it helps?

Reply:
This is probably happening because the `quansight-labs/setup-python` action returns the version number without the trailing `t` so one set of coverage results might be overriding the other. I've fixed that.

Also opend #1480 to get rif of the hypothesis warnings.